src/coreclr/tools/Common/CommandLineHelpers.cs

```diff
@@ -82,6 +82,7 @@ public static TargetOS GetTargetOS(string token)
                 "ios" => TargetOS.iOS,
                 "tvossimulator" => TargetOS.tvOSSimulator,
                 "tvos" => TargetOS.tvOS,
+                "android" => TargetOS.Linux,
```

Comment:
Is this needed given that we are replacing Android with Linux here? https://github.com/dotnet/runtime/blob/1e860a4bd3915f35d51cc079cff212322a6991e7/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets#L237-L238

We should do this substitution in a single place. Or if we believe that ilc needs to know about Android, we should go all the way and add it to TargetOS enum.